tests/databases/molecule/mongo_db/test_molecule_caching.py

```diff
@@ -0,0 +1,55 @@
+import stk
+import numpy as np
+import pymongo
+
+
+def test_put_caching():
+    database_name = '_test_put_caching'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database = stk.MoleculeMongoDb(
+        mongo_client=client,
+        database=database_name,
+    )
+    molecule = stk.BuildingBlock('CCC')
+    database.put(molecule)
+    database.put(molecule)
+
+    cache_info = database._put.cache_info()
+    assert cache_info.hits == 1
+    assert cache_info.misses == 1
+
+    database.put(
+        molecule=molecule.with_position_matrix(
+            position_matrix=np.zeros((molecule.get_num_atoms(), 3)),
+        ),
+    )
+    cache_info = database._put.cache_info()
+    assert cache_info.hits == 1
+    assert cache_info.misses == 2
+
+
+def test_get_caching():
+    database_name = '_test_get_caching'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database = stk.MoleculeMongoDb(
+        mongo_client=client,
+        database=database_name,
+    )
+    molecule = stk.BuildingBlock('CCC')
+    database.put(molecule)
```

Comment:
MAybe add the InChIKey JSON keymarker you're using here as an argument to `database.put()`. It would make it easier to understand why you're using `stk.InChIKey()` in the following `database.get` call.